src/os.ts

```diff
@@ -1,29 +1,32 @@
 enum OS {
   WIN32 = 'win32',
   WIN64 = 'win64',
-  MAC32 = 'mac32',
   MAC64 = 'mac64',
+  MAC_ARM64 = 'mac64',
   LINUX = 'linux',
+  UNSUPPORTED = 'unsupported',
 }
 
 const getOS = (): OS => {
-  const platform: string = process.platform;
-  const arch: string = process.arch;
+  const platform = process.platform;
+  const arch = process.arch;
   if (platform === 'win32') {
     return arch === 'x64' ? OS.WIN64 : OS.WIN32;
-  } else if (platform === 'darwin') {
-    return arch === 'x64' ? OS.MAC64 : OS.MAC32;
-  } else {
-    return OS.LINUX;
   }
+  if (platform === 'darwin') {
+    if (arch === 'x64') return OS.MAC64;
+    if (arch === 'arm64') return OS.MAC_ARM64;
+    return OS.UNSUPPORTED;
```

Comment:
Unsupported is added as a catch-all. To be honest, it should be added as the final return, but I'm intending on adding Linux support as well.